File: ios/baseball/baseball/Domain/Entity/Players.swift
//
//  Players.swift
//  baseball
//
//  Created by 이다훈 on 2021/05/07.
//

import Foundation

struct Players: Codable {
    private(set) var pitchers: [Pitcher]
    private(set) var batters: [Batter]
    
    init() {
        self.pitchers = []
        self.batters = []
    }
}

File: be/server/src/main/java/com/baseball/domain/player/Players.java
package com.baseball.domain.player;

import java.util.List;

public class Players {
    private final Pitchers pitchers;
    private final Batters batters;

    public Players(List<Pitcher> pitchers, List<Batter> batters) {
        this.pitchers = new Pitchers(pitchers);
        this.batters = new Batters(batters);
    }

    public List<Pitcher> getPitchers() {
        return pitchers.getPitchers();
    }

    public List<Batter> getBatters() {
        return batters.getBatters();
    }

    public Pitcher getPitcher() {
        return pitchers.getPitcher();
    }

    public Batter getBatter() {
        return batters.getBatter();
    }

    public void changePitcher() {
        pitchers.changePitcher();
    }

    public void changeBatter() {
        batters.changeBatter();
    }
}
